### PR TITLE
Sync Microsoft.Build version in analyzer template with Version.props in the workflow

### PR DESCRIPTION
**Pull request by JaynieBai**
Fixes [#9923](https://github.com/dotnet/msbuild/issues/9923)

### Context
https://github.com/dotnet/msbuild/pull/9789 added a template for custom analyzer. Microsoft.Build version specified in the template should be synced with Version.props during packing. See https://github.com/dotnet/msbuild/pull/9789/files#r1521218723 for more details.
This needs to be done after https://github.com/dotnet/msbuild/issues/9915.

### Changes Made
Create a workflow.  When there are updates in the file Version.props, the flow will be trigger and check the version. If the version in the template is different form the VersionPrefix in the Version.props. It will create a PR to sync the version between them.

### Testing
Tested in the forked repo https://github.com/JaynieBai/msbuild/actions/runs/9852188308

### CODE REVIEW

File: .github/workflows/SyncAnalyzerTemplateMSBuildVersion.yml [hunk 1]
```diff
@@ -0,0 +1,48 @@
...
+
+        # Convert the JSON content back to a string
+        $jsonString = $jsonContent | ConvertTo-Json -Depth 10
+
+        # Write the updated JSON back to the file
+        Set-Content -Path $jsonFilePath -Value $jsonString
+        Write-Output "Updated MicrosoftBuildVersion to $versionPrefix"
+
+    - name: Create Pull Request
+      uses: peter-evans/create-pull-request@v6
```
Comment (JanKrivanek) [inline .github/workflows/SyncAnalyzerTemplateMSBuildVersion.yml:42]: @baronfel - I remember you have been pointing me to a library of preapproved actions - would you be able to point again?

File: .github/workflows/SyncAnalyzerTemplateMSBuildVersion.yml [hunk 2]
```diff
@@ -0,0 +1,48 @@
...
+  Sync-version:
+    runs-on: ubuntu-latest
+
+    steps:
+    - name: Checkout repository
+      uses: actions/checkout@v3
+
+    - name: Update analyzer template version with version from Versions.props
+      shell: pwsh
+      run: |
```
Comment (JanKrivanek) [inline .github/workflows/SyncAnalyzerTemplateMSBuildVersion.yml:19]: We should have some error handling here as well - e.g. when the template config moves for some reason (e.g. renaming) - the error should surface somewhere

File: .github/workflows/SyncAnalyzerTemplateMSBuildVersion.yml [hunk 3]
```diff
@@ -0,0 +1,48 @@
...
+        Write-Output "Updated MicrosoftBuildVersion to $versionPrefix"
+
+    - name: Create Pull Request
+      uses: peter-evans/create-pull-request@v6
+      with:
+        token: ${{ secrets.GITHUB_TOKEN }}
+        commit-message: 'Update MicrosoftBuildVersion in analyzer template'
+        branch: 'update-msbuild-version-for-analyzer-template'
+        title: 'Update MicrosoftBuildVersion in analyzer template'
+        body: 'This PR updates the MicrosoftBuildVersion defaultValue in the template.json dynamically.'
```
Comment (JanKrivanek) [inline .github/workflows/SyncAnalyzerTemplateMSBuildVersion.yml:48]:
```suggestion
        body: '[Automated] Update the MicrosoftBuildVersion defaultValue in the template.json.'
```

File: .github/workflows/SyncAnalyzerTemplateMSBuildVersion.yml [hunk 4]
```diff
@@ -0,0 +1,48 @@
...
+    steps:
+    - name: Checkout repository
+      uses: actions/checkout@v3
+
+    - name: Update analyzer template version with version from Versions.props
+      shell: pwsh
+      run: |
+        # Define the paths to your XML and JSON files
+        $xmlFilePath = "eng/Versions.props"
+        $jsonFilePath = "template_feed/content/Microsoft.AnalyzerTemplate/.template.config\template.json"
```
Comment (JanKrivanek) [inline .github/workflows/SyncAnalyzerTemplateMSBuildVersion.yml:22]: Why are the slashes mixed here?

**Review by JanKrivanek — APPROVED** [review]
Overall looks good. I added couple comments for consideration

File: .github/workflows/SyncAnalyzerTemplateMSBuildVersion.yml (hunk 1 above)
Comment (baronfel) [reply .github/workflows/SyncAnalyzerTemplateMSBuildVersion.yml:42]: I don't remember that - I think the rule is that you can only use in-repo actions, or actions from the `actions/*` namespace. And all actions must be pinned to a version tag.

Comment (baronfel) [reply .github/workflows/SyncAnalyzerTemplateMSBuildVersion.yml:42]: I guess that means the 'link to library of preapproved actions' would be https://github.com/actions, then?

Comment (JanKrivanek) [reply .github/workflows/SyncAnalyzerTemplateMSBuildVersion.yml:42]: I was under impression we had yet another repo with those. But I'm very likely just confused.
Thanks for the info though!

**Comment by JaynieBai** [comment]
Fixed

File: .github/workflows/SyncAnalyzerTemplateMSBuildVersion.yml (hunk 4 above)
Comment (JaynieBai) [reply .github/workflows/SyncAnalyzerTemplateMSBuildVersion.yml]: updated

File: .github/workflows/SyncAnalyzerTemplateMSBuildVersion.yml (hunk 1 above)
Comment (JanKrivanek) [reply .github/workflows/SyncAnalyzerTemplateMSBuildVersion.yml:62]: Discussed this with @baronfel and unfrtunately we have to stick to using GH actions wither from https://github.com/actions or https://github.com/dotnet/arcade

I know arcade have couple automations creating PRs, they all seem to be through workflows. I'll reach to them to find a suggested path forward

**Review by JanKrivanek — CHANGES_REQUESTED** [review]
The change is good. We just need to move from the used GH Action. I'll research our options and get back

File: .github/workflows/SyncAnalyzerTemplateMSBuildVersion.yml (hunk 1 above)
Comment (JaynieBai) [reply .github/workflows/SyncAnalyzerTemplateMSBuildVersion.yml]: updated the task with the GitHub actions. Please help review again

**Review by JanKrivanek — APPROVED** [review]
Looks good
Thank you!

**Review by YuliiaKovalova — APPROVED** [review]

**Pull request merged**